Sources/ManageMembergroups.php

```diff
@@ -797,6 +825,31 @@ function EditMembergroup()
 	// A form was submitted with the new membergroup settings.
 	elseif (isset($_POST['save']))
 	{
+		// Are we inheriting? Account groups can't inherit from character groups, and vice versa.
+		$current_group_is_character = StoryBB\Model\Group::is_character_group((int) $_REQUEST['group']);
+		if (isset($_POST['group_inherit']) && $_POST['group_inherit'] != -2)
```

Comment:
negative two?

Reply:
There's a bunch of ways that a group can be inherited or copied for permissions. Positive numbers indicate a group to inherit from, including -1 for guests. -2 relates to the specific type of inheritance here where we don't copy from the group but try to reuse it. It's weird and ugly and could do with a rethink sometime.